tensorflow/core/kernels/transpose_op.cc

```diff
@@ -133,7 +133,16 @@ void TransposeOp::Compute(OpKernelContext* ctx) {
   OP_REQUIRES(ctx, TensorShapeUtils::IsVector(perm.shape()),
               errors::InvalidArgument("perm must be rank 1, got shape ",
                                       perm.shape().DebugString()));
-
+  // Validation for negative values inside `perm`.
+  auto perm_vector = perm.vec<int>();
+  for (int i = 0; i < perm.dim_size(0); i++) {
```

Comment:
Thanks for the changes! Some minor comments to adhere to the c++ style guide.
Please use the prefix form (++i) - https://google.github.io/styleguide/cppguide.html#Preincrement_and_Predecrement

Reply:
Sure.Thanks!